src/svcutilcore/src/Microsoft/Tools/ServiceModel/SvcUtil/XmlSerializerGenerator.cs

```diff
@@ -124,21 +122,142 @@ private List<Type> CollectXmlSerializerTypes(Assembly assembly, List<XmlMapping>
                         ToolConsole.WriteWarning(SR.Format(SR.WrnUnableToLoadContractForSGen, contractType, errorMessage));
                     };
 
-            foreach (ContractDescription contract in contractLoader.GetContracts())
+            Type contractDescriptionType = Tool.SMAssembly.GetType("System.ServiceModel.Description.ContractDescription");
+            if (contractDescriptionType == null)
             {
-                types.Add(contract.ContractType);
-                foreach (OperationDescription operation in contract.Operations)
+                ToolConsole.WriteError($"Not found type System.ServiceModel.Description.ContractDescription in {Tool.SMAssembly.FullName}");
+                throw new ToolRuntimeException();
+            }
+
+            PropertyInfo contractTypeProperty = contractDescriptionType.GetProperty("ContractType");
+            if (contractTypeProperty == null)
+            {
+                ToolConsole.WriteError($"Not found property ContractType in type {contractDescriptionType}");
+                throw new ToolRuntimeException();
+            }
+
+            Type operationsType = Tool.SMAssembly.GetType("System.ServiceModel.Description.OperationDescriptionCollection");
+            if (operationsType == null)
+            {
+                ToolConsole.WriteError($"Not found type System.ServiceModel.Description.OperationDescriptionCollection in {Tool.SMAssembly.FullName}");
+                throw new ToolRuntimeException();
+            }
+
+            Type xmlSerializerOperationBehaviorType = Tool.SMAssembly.GetType("System.ServiceModel.Description.XmlSerializerOperationBehavior");
+            if (xmlSerializerOperationBehaviorType == null)
+            {
+                ToolConsole.WriteError($"Not found type System.ServiceModel.Description.XmlSerializerOperationBehaviorType in {Tool.SMAssembly.FullName}");
+                throw new ToolRuntimeException();
+            }
+
+            Type operationType = Tool.SMAssembly.GetType("System.ServiceModel.Description.OperationDescription");
+            if (operationType == null)
+            {
+                ToolConsole.WriteError($"Not found type System.ServiceModel.Description.OperationDescription in {Tool.SMAssembly.FullName}");
+                throw new ToolRuntimeException();
+            }
+
+            PropertyInfo getBehaviors = operationType.GetProperty("Behaviors");
+            if (getBehaviors == null)
+            {
+                ToolConsole.WriteError($"Not found method get_Behaviors in type {operationType}");
+                throw new ToolRuntimeException();
+            }
+
+            Type KeyedByTypeCollection = Tool.SMAssembly.GetType("System.Collections.Generic.KeyedByTypeCollection`1");
+            if (KeyedByTypeCollection == null)
```

Comment:
There are a couple of variables which are incorrectly named. Variables should begin with lowercase letter and you should probably suffix with what it is such as Type. IOperationBehavior should be renamed. Also getBehaviors should probably be getBehaviorsProperty and find should be findMethod etc.